src/EditorFeatures/CSharpTest/CodeActions/ConvertLinq/ConvertForEachToLinqQueryTests.cs

```diff
@@ -0,0 +1,2842 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.ConvertLinq
+{
+    public class ConvertForEachToLinqQueryTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery.CSharpConvertForEachToLinqQueryProvider();
+
+        #region Query Expressions
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForEachToQuery)]
+        public async Task QueryForForWhere()
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+class Query
+{
+    public IEnumerable<int> void Main(string[] args)
+    {
+        List<int> c1 = new List<int>{1, 2, 3, 4, 5, 7};
+        List<int> c2 = new List<int>{10, 30, 40, 50, 60, 70};
+        [|foreach (var x1 in c1)
```

Comment:
things like var @object in c1 work?

Reply:
added

---
In reply to: [195851657](https://github.com/dotnet/roslyn/pull/27130#discussion_r195851657) [](ancestors = 195851657)